### DISCUSSION TITLE
Wrong SMARTS pattern in O-benzyl deprotection reaction

### DISCUSSION THREAD

**Discussion by marcobICR** [rdkit/rdkit#7982] [Development]
Hi everyone, 

I have noticed that in the benzyl deprotection reaction the SMARTS pattern is wrong, matching phenylethers instead:

```python
import rdkit
from rdkit import Chem
from rdkit.Chem import Draw, rdDeprotect

mol = Chem.MolFromSmiles("Nc1ccccc1Oc1ccccc1")
```

This is the mol:
![image](https://github.com/user-attachments/assets/0899ab60-e623-4aa1-9674-5d40a0c2e5b9)

```python
Draw.MolsToGridImage(
    mols = [rdDeprotect.Deprotect(mol = mol, deprotections = [dep]) for dep in rdDeprotect.GetDeprotections()],
    legends = [dep.full_name for dep in rdDeprotect.GetDeprotections()],
    molsPerRow = 7
)
```
This is the output:
![image](https://github.com/user-attachments/assets/2601c659-18f2-49d7-b8fb-5cd0cda83ca3)

This compound should not be deprotected by the benzyl reaction at all, since this is a phenyl ether.
Looking at the [SMARTS in the code:](https://github.com/rdkit/rdkit/blob/b16b6026e8d58ddba73a7ba6b24f5c0412fd4951/Code/GraphMol/Deprotect/Deprotect.cpp#L53)

```
      {"alcohol", "[O;!$(*C(=O)):1]c1[c;H1][c;H1][c;H1][c;H1][c;H1]1>>[O;H1:1]",
       "Bn", "benzyl", "NOc1ccccc1>>ON"}
```

It looks like the SMARTS pattern is wrong and it's missing a CH2 group between the O and the phenyl group.
Can someone confirm that this is indeed uncorrect?

**Comment by greglandrum** [comment]
@marcobICR I agree that this looks wrong. Could you please create an issue for this?

**Comment by marcobICR** [comment]
Hi @greglandrum, thanks. I have opened the issue, please let me know if I should add anything else.
